python/lsst/daf/butler/core/utils.py

```diff
@@ -203,3 +203,87 @@ def __call__(cls):  # noqa N805
         if cls not in cls._instances:
             cls._instances[cls] = super(Singleton, cls).__call__()
         return cls._instances[cls]
+
+
+class ConnectedSet:
```

Comment:
Does it need a `__len__` to make it properly look like a `collections.Set`?

Reply:
Probably. I'll add it. And perhaps some of the other methods too. But I'm not sure supporting anything other than iteration, length and perhaps `union` is worth the effort at the moment.